src/build.rs

```diff
@@ -0,0 +1,122 @@
+use std;
+use super::{
+	CREATE_SYMBOL,
+	CALL_SYMBOL,
+	optimize,
+	pack_instance,
+	ununderscore_funcs,
+	externalize_mem,
+	shrink_unknown_stack,
+	inject_runtime_type,
+	PackingError,
+	OptimizerError
+};
+use parity_wasm;
+use parity_wasm::elements;
+
+#[derive(Debug)]
+pub enum Error {
+	Encoding(elements::Error),
+	Packing(PackingError),
+	NoCreateSymbolFound,
+	Optimizer,
+}
+
+impl From<OptimizerError> for Error {
+	fn from(_err: OptimizerError) -> Self {
+		Error::Optimizer
+	}
+}
+
+impl From<PackingError> for Error {
+	fn from(err: PackingError) -> Self {
+		Error::Packing(err)
+	}
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum Target {
```

Comment:
I don't think Target is better name that SourceTarget, since SourceTarget outlines that it is the target which we take as a source to process further

Reply:
sure. 
But because now it lives in `build.rs`, which don't care know about "files" and now it doesn't require this context clarification, IMO

Reply:
But the logic retained
Why not call it "source" then?